graylog2-shared/src/main/java/org/graylog2/shared/rest/NodeIdResponseFilter.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.shared.rest;
+
+import org.graylog2.plugin.system.NodeId;
+
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import java.io.IOException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class NodeIdResponseFilter implements ContainerResponseFilter {
+    private final NodeId nodeId;
+
+    @Inject
+    public NodeIdResponseFilter(NodeId nodeId) {
+        this.nodeId = checkNotNull(nodeId);
```

Comment:
Yeah, it's quite defensive but it prevents stupid me from initializing the class manually in a test with `null` instead of a `NodeId` instance.
![](https://33.media.tumblr.com/083070869448cdf23833058c5006aab8/tumblr_np90txzvfJ1u8r5tto1_540.gif)